python_modules/dagster/dagster/_core/definitions/source_asset.py

```diff
@@ -126,6 +161,46 @@ def io_manager_def(self) -> Optional[IOManagerDefinition]:
             self.resource_defs.get(io_manager_key) if io_manager_key else None,
         )
 
+    @public  # type: ignore
+    @property
+    def op(self) -> OpDefinition:
+        check.invariant(
+            isinstance(self.node_def, OpDefinition),
+            "The NodeDefinition for this AssetsDefinition is not of type OpDefinition.",
+        )
+        return cast(OpDefinition, self.node_def)
+
+    @public
+    @property
+    def is_versioned(self) -> bool:
+        return self.node_def is not None
+
+    @property
+    def node_def(self):
+        """Op that generates observation metadata for a source asset."""
+        if self.observe_fn is None:
+            return None
+        else:
+            observe_fn = self.observe_fn
```

Comment:
Is this line necessary? Why not just use `self.observe_fn` below?

Reply:
This is required to create a closure for the observe function-- mypy flags it otherwise. While I don't _think_ it would break using `self.observe_fn` directly, the type-checker correctly observes that `self.observe_fn`, being a nullable property, might return `None` when it's called at some unpredictable point inside the `compute_fn`.